docs/usage/macros.rst

```diff
@@ -42,3 +42,28 @@ Example:
 
   spl_add_source(src/led_driver_main.c)
   spl_add_source(src/led_driver_control.c COMPILE_OPTIONS "-w")
+
+
+
+spl_add_compile_options
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Add a compile option to the list of sources to be compiled.
```

Comment:
It is actually not one compile option but a arbitrary number. And it is not added to "the list of sources". We define a pattern here. Please rephrase it and be more precise. A user has to understand it while reading without implementation knowledge. The information, that the given pattern is a path relative to the current CMake file is also worth mentioning.